for revise (#116)
fulfillmenttools/fulfillmenttools-api-reference: doc/orders/orders.adoc

== Orders
image::doc/orders/DocOrdersOverview.png[align="center"]

The distributed order management systems (DOMS) is a backend component that orchestrates the incoming orders from your online-store and decides where and how the order can be fulfilled best using several routing rules to meet your consumers needs best.

An order is a collection of the demanded items of a consumer plus some extra information, which should be fulfilled.
It is also a typical entry point to the operational processes within the fulfillmenttools platform: An order is created after the consumer went through the checkout in the online shop. The following paragraph tells about the abilities and features that are enclosed within this entity.

=== Default behaviour

> An order, that enters the system, is typically mapped to one or more <<Pickjobs>> and dispatched to <<Facilities>> for fulfillment. Routing rules could be listing based, stockbased, location based, etc. according the routing rules currently active in the system.

Reviewer: ```suggestion
An order is typically mapped to one or more <<Pickjobs>> and dispatched to <<Facilities>> for fulfillment after it has been supplied to the platform. There are Routing rules present in the system, that could be configured in order to meet certain expectations towards for example stock or geolocation based distribution. These rules apply by default to every new order, that is entering the system.
```

> This default behaviour can be bypassed by certain information   in the order, namely the `deliverypreferences` which is a simple field in the order. If you provide an order omitting this field it will default to:

Reviewer: ```suggestion
This default behaviour can be bypassed by certain information in the order message, namely the `deliverypreferences` field. If you provide an order not having this field the following default will be added:
```

[source,curl]
----
{
    ...
    "deliverypreferences" : [
        "shipping": {
             "servicetype": "BEST_EFFORT"
            }
    ]
    ...
}
----

> Default setting for delivery preference of an order is set to shipping and different types of service type is considered that also "BEST_EFFORT" is the default one.

Reviewer: ```suggestion
The order will be **shipped** by a carrier using a **best effort** servicetype. 
```

=== Providing preferred delivery types per order

> With fulfillmenttools platform you can currently provide

Reviewer: ```suggestion

```

> Consumer has the option to choose desired delivery approach, whether pick up the order by own from store or request it to be delivered to consumer address. Moreover, in click and collect option, consumer has the option to choose between possible facilities.

Reviewer: ```suggestion
With fulfillmenttools platform you can currently provide a certain set of parameters that are used within the process after the picking. The order can carry the following information about the desired delivery approaches:
* the picked goods are handed over in person at the store where the pickjob was fulfilled (**Click & Collect**)
* the picked items are packed and sent via a parcel carrier to the consumer (**Delivery**, _**default**_)
```

[source,curl]
----
"deliveryPreferences": {
        "collect": [
            {

>              "facilityRef": "50678"  // The facility, consumer has chosen to hand over the order

Reviewer: ```suggestion
             "facilityRef": "28MB9nc4njINUx9rDrCU"  // The facility which the consumer has chosen to hand over the order
```

            }
        ]
}
----

> These information will be available in related pickjob :

Reviewer: This I would move to the pickjobs section.

[source,curl]
----
    "deliveryinformation": {
        "channel": "COLLECT",
        "details": {
            "collect": {
                 "identifier": "Stefan Bauer"  // Identity of the consumer
            },
            "shipping": {
              // Contains address of the consumer
            }
        }
    }
----
By Default channel field of the delivery information of a pick job is set to "SHIPPING" but in Click & Collect option, it is "COLLECT"

> === Injecting information into fulfillmenttools solutions

Reviewer: ```suggestion

```

> * Possibility to provide information about an item (displayed in the Instore-Fulfillment App)

Reviewer: ```suggestion

```

== Orders
image::doc/orders/DocOrdersOverview.png[align="center"]

> An order is a collection of the demanded items of a consumer to be fulfilled. It is also the typical entry point to the fulfillment process within the fulfillmenttools platform: An order is created after the consumer went through the checkout in the online shop. Consumer has the option to choose his/her desired delivery approach, whether pick up the order by her/his self from store or request it to be delivered to her/his address. Moreover, in click and collect option, consumer has the option to choose between possible facilities.

Reviewer: ```suggestion
An order is a collection of the demanded items of a consumer to be fulfilled. It is also the typical entry point to the fulfillment process within the fulfillmenttools platform: An order is created after the consumer went through the checkout in the online shop. 

Some shops offer the possibility to collect the ordered items at a specific store / facility. This preference can be supplied to the platform with the order by using `deliverypreferences`. ```

> 

Reviewer: Also I think we should tell about the priority and types in the deliverypreferences - so that depending on the configuration of your platform it could be, that the order is splitted and a part of it is being sent while another part can be picked up at a store.

Or maybe this is already too much information?

Also: A curl example on how to create an order would be cool I guess.
